test/libsolidity/ABIDecoderTests.cpp

```diff
@@ -332,6 +369,7 @@ BOOST_AUTO_TEST_CASE(decode_function_type)
 			}
 		}
 	)";
+	// TODO validation
```

Comment:
This comment can be removed, since we have a test case for this below, right?

Reply:
Right!